### DISCUSSION TITLE
FreeBSD support?

### DISCUSSION THREAD

**Discussion by tschettervictor** [bluenviron/mediamtx#3122] [Questions] [answered]
### Question

Is FreeBSD supported? Can’t seem to run the binary.

**Comment by aler9** [comment, marked as answer]
If you need binaries for operating systems that are not officially supported, you can build the server from source by following instructions in the [README](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#compile-from-source).

**Reply by tschettervictor** [reply]
Excellent. Got it working nicely. Will post an rc service script shortly.

Any reason authentication would fail?

Does it need to be hashed or something?

**Comment by tschettervictor** [comment]
Also, is it possible to set a user to only be able to view the streams? Currently I have a user set with action "read" but when I try to view the stream, it fails with auth error. Only when I enable the user to also "publish" does it work.

Using HLS to view, and rtmp to stream

**Comment by tschettervictor** [comment]
I actually seem to have got this to work somehow, not exactly sure how, but playing around with the config file did the trick.

[a comment by an automated account is not shown]
